addons/test_mail_full/models/test_mail_models.py

```diff
@@ -49,6 +49,30 @@ def _sms_get_number_fields(self):
         return ['phone_nbr', 'mobile_nbr']
 
 
+class MailTestSMSOptout(models.Model):
+    """ Model using blacklist mechanism and a hijacked opt-out mechanism for
+    mass mailing features. """
+    _description = 'SMS Mailing Blacklist / Optout Enabled'
+    _name = 'mail.test.sms.bl.optout'
+    _inherit = ['mail.thread.phone']
+    _order = 'name asc, id asc'
+
+    name = fields.Char()
+    subject = fields.Char()
+    email_from = fields.Char()
+    phone_nbr = fields.Char()
+    mobile_nbr = fields.Char()
+    customer_id = fields.Many2one('res.partner', 'Customer')
+    opt_out = fields.Boolean()
```

Comment:
I'm guessing you can't just add a "opt_out" field on the "mail.test.sms" model but I'm not sure why?

I see you already have duplicated models for "MailTestSMS" and "MailTestSMSBL".

Reply:
I did it Rco style, aka one model per "new feature" to test. That way we test both simple models and complex models, not just complex models.